Update “hack” to catch Duration.Inf.toMillils now throwing IllegalArgumentException.

austinpernell opened the pull request:
Using any version of Play 2.6.X the default value of play.http.session.maxAge of null is being converted to Duration.Inf. https://github.com/playframework/playframework/commit/84c883d3865266da10706eddd71026821912000c#diff-4378e6503263d81d7869f6056b31e6b1L1037

```
private[this] def fail(what: String) = throw new IllegalArgumentException(s"$what not allowed on infinite Durations")
    final def length: Long    = fail("length")
    final def unit: TimeUnit  = fail("unit")
    final def toNanos: Long   = fail("toNanos")
    final def toMicros: Long  = fail("toMicros")
    final def toMillis: Long  = fail("toMillis")
    final def toSeconds: Long = fail("toSeconds")
    final def toMinutes: Long = fail("toMinutes")
    final def toHours: Long   = fail("toHours")
    final def toDays: Long    = fail("toDays")
```

IllegalArgumentException is now being thrown where ConfigException.Null may have been thrown in the past.

Another route to go instead of using this hack could be to use getOptional[Long] instead of getMilliseconds since it uses getOptional[Duration].map(_.toMillis). This would prevent the use of a Duration in the maxAge field when using this library, but may be less brittle going forward.

I apologize if this PR isn't fully complete. This is my first open source PR, and can definitely make changes to documentation that I may have missed, or make changes to version numbers and resubmit.

austinpernell committed:
Update “hack” to catch Duration.Inf.toMillils now throwing IllegalArgumentException.

austinpernell committed:
Think this is correct for the version number change

pauldijou commented:
Hey, thanks for the work. Not sure about your code, we do not want to swallow every possible `IllegalArgumentException` that could happen for any other reason. Could you try to comment your code and setup a failing test so I can easily reproduce? Already existing tests are here: https://github.com/pauldijou/jwt-scala/tree/master/play/src/test/scala

When done, try using `getMillis` here since `getMilliseconds` is now deprecated : https://github.com/pauldijou/jwt-scala/blob/master/play/src/main/scala/JwtSession.scala#L102.

If not solving it, I will give it a closer look.

austinpernell commented:
Sorry I didn't include a test. They all ran fine until I got:` [error] (playLegacy/test:testGrouping) java.lang.NoClassDefFoundError: $5a39c98268d8f521595f$$anonfun$groupPlayTest$1`
and never could manage to get beyond that. 

Since the Play Framework is actually trying to gather a FiniteDuration, instead of just a Duration (that is returning InfiniteDuration, hence the error), how about just converting to getOptional[FiniteDuration]? 
https://github.com/playframework/playframework/blob/2.6.x/framework/src/play/src/main/scala/play/api/http/HttpConfiguration.scala#L206

The code would then just become:
`key => Play.maybeApplication.flatMap(_.configuration.getOptional[FiniteDuration](key).map(_.toMillis))`
and the catching IllegalArgumentException could be removed. That's probably much cleaner. I'll resubmit that, and keep attempting to run the tests so I can add checks in that way too.

austinpernell committed:
PR cleanup.

Clean up code by not catching all IllegalArgumentExceptions but rather parse the field in a more appropriate manner.

austinpernell committed:
Tests for maxAge being null.

austinpernell commented:
@pauldijou I think these tests should validate my findings, I was finally able to get them running.

pauldijou commented:
Thanks a lot for failing test. I finally decided to go with checking if the key is present or not. Is that ok with you? https://github.com/pauldijou/jwt-scala/commit/1d466abc6e903a0d724465a6259e75892f2f4b04

austinpernell commented:
No problem! Sure, that will definitely prevent any errors from being thrown if the value is null. Thanks for taking the time to review and merge my PR, and thanks for your libraries!

pauldijou commented:
Cool. Should be released in 0.14.1
